os/hal/ports/STM32/STM32F0xx/hal_lld.c

```diff
@@ -60,7 +60,15 @@ static void hal_lld_backup_domain_init(void) {
   PWR->CR |= PWR_CR_DBP;
 
   /* Reset BKP domain if different clock source selected.*/
-  if ((RCC->BDCR & STM32_RTCSEL_MASK) != STM32_RTCSEL) {
+  if (
+#if STM32_LSE_ENABLED
+      (
+       fomeLseCounter >= FOME_STM32_LSE_MAX_WAIT
```

Comment:
This errors as the declaration is later.

Also, this idea doesn't work; I think `fomeLseCounter` would have to be `static` for it to work?

Reply:
> Also, this idea doesn't work; I think `fomeLseCounter` would have to be `static` for it to work?

seems to build ... does it have the effect I'm hoping for?  I assume this `init` function is called on power restore and such?  maybe `static` doesn't survive that ...

https://github.com/nmschulte/ChibiOS/compare/nms/lse-max_wait-bkp-domain-reset-v20.3.x...nms/lse-max_wait-test-v20.3.x#diff-2ce335069c06af5a22b9f80d71f4e497c5cab4c2f633afcfc55ed1f015dae218